File: Codeforces/Rudolf_and_121.py
def makeZero(A, n):
    xor = 0
    isZero = True

    for i in range(n):
        if A[i] > 0:
            isZero = False
        xor ^= A[i]

    if isZero:
        return "YES"
    elif xor == 0:
        return "YES"
    else:
        return "NO"

# Example usage
t = int(input())
for _ in range(t):
    n = int(input())
    A = list(map(int, input().split()))
    print(makeZero(A, n))

File: Codeforces/Rudolf_and_121.cpp
#include <iostream>
#include <vector>

using namespace std;

string makeZero(vector<int>& A, int n) {
    int xorSum = 0;
    bool isZero = true;

    for (int i = 0; i < n; ++i) {
        if (A[i] > 0) {
            isZero = false;
        }
        xorSum ^= A[i];
    }

    if (isZero || xorSum == 0) {
        return "YES";
    } else {
        return "NO";
    }
}

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;

        vector<int> A(n);
        for (int i = 0; i < n; ++i) {
            cin >> A[i];
        }

        cout << makeZero(A, n) << "\n";
    }

    return 0;
}
